[docs] Add documentation for `eas build:run` command for Android Emulator and iOS Simulator (#21112)
expo/expo: docs/pages/build-reference/apk.mdx

### Emulator ("Virtual device")

> If you haven't installed or run an Android Emulator before, follow the [Android Studio emulator guide](/workflow/android-studio-emulator) before proceeding.

Once your build is completed, the CLI will prompt you to automatically download and install it on the Android Emulator. When prompted, press <kbd>Y</kbd> to directly install it on the emulator.

In case you have multiple builds, you can also run the `eas build:run` command at any time to download a specific build and automatically install it on the Android Emulator:

<Terminal cmd={['$ eas build:run -p android']} />

The command also shows a list of available builds of your project. You can select the build to install on the emulator from this list. Each build in the list has a build ID, the time elapsed since the build creation, the build number, the version number, and the git commit information. The list also displays invalid builds if a project has any.

For example, the image below lists the build of a project:

<ImageSpotlight
  alt="Running eas build:run command shows a list of available builds in a project."
  src="/static/images/eas-build/eas-build-run-on-android.png"
/>

> When the build's installation is complete, it will appear on the home screen. Tap it to run it, and in the terminal window, start the development server using `npx expo start --dev-client`.

Reviewer: ```suggestion
When the build's installation is complete, it will appear on the home screen. If it's a development build, open a terminal window and start the development server using `npx expo start --dev-client`.
```

Reviewer: you might use this to preview your app (internal distribution)
